src/rocker/templates/user_snippet.Dockerfile.em

```diff
@@ -1,3 +1,6 @@
+ARG USERNAME=@(name)
+ARG HOMEDIR=@(dir)
+
```

Comment:
These are not the only parameters to the Dockerfile, if we're doing this we should cover them all. The uid and gid are also important if you change the user.

Reply:
My specific user case was to quickly create my own Dockerfile for another project and after seeing the dumped one in the rocker output I guessed it would be a good addition to other people.
It would be extended for many common use cases in which we can't use rocker in a docker compose but we could create our own images with the dockerfiles generated by rocker. 

I really think that the Dockerfiles that rocker internally uses are a nice reference for other projects needing nvidia, user, x11, etc in it projects.

Maybe I could implement an option in rocker to save the Docker files instead of building itself.

Reply:
The main design goal of rocker is to provide dynamic content that is not possible to capture in a container image distributed by a project. (Aka it's not appropriate to bake in the $USERNAME to an image. Which is exactly why rocker detects and generates the content dynamically. I have ideas about how to enable baking in cached content, but I think that trying to make the dynamically generated Dockerfiles exportable will just lead to people hitting unexpected errors. For example as you've proposed this will break unless you happen to have the same UID.

Reply:
I understand. Thank you so much for taking the time to explain it. Best!